File: lib/models/service_request.dart
enum ServiceRequestType {
  emergency,
  maintenance,
  cleaning,
  equipment,
  other,
}

enum ServiceRequestStatus {
  pending,
  accepted,
  inProgress,
  completed,
  cancelled,
}

class ServiceRequest {
  final String id;
  final String userId;
  final ServiceRequestType type;
  final String title;
  final String description;
  final String location;
  ServiceRequestStatus status;
  final DateTime requestedAt;
  DateTime? acceptedAt;
  DateTime? completedAt;
  String? assignedTo;
  String? notes;
  List<String>? images;
  final bool isUrgent;
  DateTime? scheduledDate;
  DateTime? scheduledTime;

  ServiceRequest({
    required this.id,
    required this.userId,
    required this.type,
    required this.title,
    required this.description,
    required this.location,
    this.status = ServiceRequestStatus.pending,
    required this.requestedAt,
    this.acceptedAt,
    this.completedAt,
    this.assignedTo,
    this.notes,
    this.images,
    this.isUrgent = false,
    this.scheduledDate,
    this.scheduledTime,
  });

  String get typeLabel {
    switch (type) {
      case ServiceRequestType.emergency:
        return 'Яаралтай';
      case ServiceRequestType.maintenance:
        return 'Засвар';
      case ServiceRequestType.cleaning:
        return 'Цэвэрлэгээ';
      case ServiceRequestType.equipment:
        return 'Тоног төхөөрөмж';
      case ServiceRequestType.other:
        return 'Бусад';
    }
  }

  String get statusLabel {
    switch (status) {
      case ServiceRequestStatus.pending:
        return 'Хүлээгдэж байна';
      case ServiceRequestStatus.accepted:
        return 'Хүлээн авсан';
      case ServiceRequestStatus.inProgress:
        return 'Явж байна';
      case ServiceRequestStatus.completed:
        return 'Дууссан';
      case ServiceRequestStatus.cancelled:
        return 'Цуцлагдсан';
    }
  }
}

File: backend/models/ServiceRequest.js
// Үйлчилгээний хүсэлтийн модель
// Энэ файл нь үйлчилгээний хүсэлтийн мэдээллийг хадгалах бүтцийг тодорхойлдог

const mongoose = require('mongoose');

// Үйлчилгээний хүсэлтийн schema
const serviceRequestSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User', // User моделд холбох
    required: true // Заавал оруулах - хүсэлт гаргасан хэрэглэгч
  },
  type: {
    type: String,
    enum: ['emergency', 'maintenance', 'cleaning', 'equipment', 'other'], // Зөвхөн эдгээр төрлүүд
    required: true // Заавал оруулах - хүсэлтийн төрөл
    // emergency: яаралтай, maintenance: засвар, cleaning: цэвэрлэгээ, equipment: тоног төхөөрөмж, other: бусад
  },
  title: {
    type: String,
    required: true // Заавал оруулах - хүсэлтийн гарчиг
  },
  description: {
    type: String,
    required: true // Заавал оруулах - дэлгэрэнгүй тайлбар
  },
  location: {
    type: String,
    required: true // Заавал оруулах - байршил
  },
  status: {
    type: String,
    enum: ['pending', 'accepted', 'inProgress', 'completed', 'cancelled'], // Зөвхөн эдгээр статусууд
    default: 'pending' // Анхдагч статус
    // pending: хүлээгдэж буй, accepted: хүлээн авсан, inProgress: хийгдэж буй, completed: дууссан, cancelled: цуцлагдсан
  },
  requestedAt: {
    type: Date,
    default: Date.now // Хүсэлт гаргасан огноо
  },
  acceptedAt: {
    type: Date // Хүлээн авсан огноо
  },
  completedAt: {
    type: Date // Дууссан огноо
  },
  assignedTo: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Technician' // Technician моделд холбох - хүсэлтэд томилогдсон засварчин
  },
  notes: {
    type: String // Нэмэлт тэмдэглэл
  },
  images: [{
    type: String // Зургийн URL-уудын массив
  }],
  isUrgent: {
    type: Boolean,
    default: false // Яаралтай эсэх
  },
  scheduledDate: {
    type: Date // Төлөвлөсөн огноо
  },
  scheduledTime: {
    type: Date // Төлөвлөсөн цаг
  }
}, {
  timestamps: true // createdAt, updatedAt талбаруудыг автоматаар нэмэх
});

module.exports = mongoose.model('ServiceRequest', serviceRequestSchema);
